crates/nu-command/src/formats/from/nuon.rs

```diff
@@ -74,8 +74,29 @@ impl Command for FromNuon {
         input: PipelineData,
     ) -> Result<nu_protocol::PipelineData, ShellError> {
         let head = call.head;
+        let input_span = input.span().unwrap_or(head);
         let (string_input, metadata) = input.collect_string_strict(head)?;
 
+        // check if input contains parser keyword like `register xxx`, or `module xx`
+        // or `let xx = xxx` or something else.
+        // If true, it's absolutely not a valid nuon data, return error early.
+        // So we can avoid panic during parsing state.
+        let parser_keyword_names = _engine_state.parser_keyword_names();
```

Comment:
Don't know how pressing the need to optimize that yet. But the available keywords should be known at the beginning of the session and never change. Or rather a nuon file should always be valid. so it shouldn't matter if you try to open it with a full command set or a limited context. Thus the list of keywords should be known at compile time.

Reply:
> Or rather a nuon file should always be valid. 

I don't think `nuon` file is the same to `nu` file, `nu` file contains data as well as code, but `nuon` file contains only data.  So when we open `nu` file, it can be failed because it contains code.